packages/experimental/AppearanceAdditions/ios/FRNAppearanceAdditions.m

```diff
@@ -145,18 +148,23 @@ - (void)appearanceChanged:(NSNotification *)notification {
         NSString *horizontalSizeClass = RCTHorizontalSizeClassPreference(traitCollection);
         NSString *userInterfaceLevel = RCTUserInterfaceLevelPreference(traitCollection);
         NSString *accessibilityContrastOption = RCTAccessibilityContrastPreference(traitCollection);
-
-        if (![horizontalSizeClass isEqualToString:_horizontalSizeClass] ||
-            ![userInterfaceLevel isEqualToString:_userInterfaceLevel] ||
-            ![accessibilityContrastOption isEqualToString:_accessibilityContrastOption]) {
-            _horizontalSizeClass = horizontalSizeClass;
-            _userInterfaceLevel = userInterfaceLevel;
-            _accessibilityContrastOption = accessibilityContrastOption;
+        
+        NSNumber *rootTag = [[[notification object] contentView] reactTag];
```

Comment:
I think we should guard that the selector exists here, or else we might get a runtime crash.

Reply:
I prefer class verification, since it verifies that the selectors will also return the types you think (while pure duck-typed selector sniffing does not):

```objc
if (![[notification object] isKindOfClass:[RCTRootView class]]) {
    return;
}
RCTRootView *rootView = [notification object];
NSNumber *rootTag = [rootView rootTag];
```

Reply:
Just updated - I thought that the verification for trait collection and rootTag should be the same, so I also changed the behaviour so that if the verification for trait collection fails we just return as well. 

Some questions:
- I'm unclear what's happening in the `UITraitCollection *traitCollection = [[notification userInfo] valueForKey:...]` line. At this point, if the right side doesn't return a UITraitCollection pointer, shouldn't this cause some kind of error instead of at the check on the next line where we explicitly check the class? Thinking about it more I would have expected this to result in a compiler error. Testing locally, if I change `UITraitCollection *traitCollection` to something clearly wrong like `UIView *traitCollection` everything actually still works
- I'm wondering if we should also check that contentView and reactTag are both non-nil